lib/solargraph/shell.rb

```diff
@@ -140,12 +141,13 @@ def uncache *gems
     # @return [void]
     def gems *names
       api_map = ApiMap.load('.')
+      workspace = api_map.workspace
       if names.empty?
         Gem::Specification.to_a.each { |spec| do_cache spec, api_map }
         STDERR.puts "Documentation cached for all #{Gem::Specification.count} gems."
       else
         names.each do |name|
-          spec = Gem::Specification.find_by_name(*name.split('='))
+          spec = workspace.find_gem(*name.split('='))
```

Comment:
Encapsulating this into Gemspecs so we can fix issues there.  For instance, today we can end up caching the wrong version of the gem because we aren't asking for the version contained in the bundle.